core/src/test/java/org/apache/druid/storage/local/LocalFileStorageConnectorTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.storage.local;
+
+import org.apache.druid.storage.StorageConnector;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.UUID;
+
+public class LocalFileStorageConnectorTest
+{
+  @Rule
+  TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private File tempDir = temporaryFolder.newFolder();
+  private StorageConnector storageConnector = new LocalFileStorageConnectorProvider(tempDir).get();
+
+  public LocalFileStorageConnectorTest() throws IOException {}
```

Comment:
This is not needed, and will fail both checkstyle and IntelliJ inspections, sadly.

Reply:
as tempFolder.newFolder() throws an exception, the constructor is maked with this but I do see that our checkstyle would most likely fail. 
Adjusted the code to make it pass.